<a href="https://colab.research.google.com/github/Ansu-Devog/Data-analysis-Internship-Cognifyz-/blob/main/Level%202/Restaurant_Insights.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Project Name - Restaurant Insights**

##### **Project Type**    - EDA
##### **Industry**    - Cognifyz Technologies
##### **Contribution**    - Individual
##### **Member Name -** Ansuman Samanta
##### **Level -** 2

# **Project Summary**

In the second phase of my Data Analysis internship at Cognifyz Technologies, I embarked on a journey to enhance our understanding of the restaurant dataset through a series of focused tasks. This level 2 revolved around three key objectives:

**Task 1: Table Booking and Online Delivery Analysis**

- Calculated the percentage of restaurants offering table booking and online delivery services.
- Conducted a comparative analysis of average ratings for restaurants with and without table booking.
- Explored the availability of online delivery options across various price ranges, revealing valuable insights into customer preferences.

**Task 2: Price Range Analysis**

- Determined the most common price range among all restaurants.
- Computed the average rating for each price range, helping identify high-performing price segments.
- Identified the color associated with the highest average rating, simplifying the identification of top-rated price ranges.

**Task 3: Feature Engineering**

- Employed advanced feature engineering techniques to extract additional insights from existing columns.
- Generated new features like "Has Table Booking" and "Has Online Delivery" by encoding categorical variables, enhancing the dataset's intelligence.

This level of the project aimed to elevate our analysis by uncovering essential details related to table booking, delivery services, pricing, and feature engineering. The insights gained will contribute significantly to our data-driven decision-making process and project success.

# **Github**

Link - https://github.com/Ansu-Devog/Data-analysis-Internship-Cognifyz-/tree/5021a3d00a065ef375d7bea0eec5c8a5a35ff241/Level%202

# **Problem Statement**

In Level 2 of the Cognifyz Data Analysis Internship, the objective is to further enrich and refine our understanding of the restaurant dataset through three distinct tasks:

**Task 1: Table Booking and Online Delivery**

- Calculate the percentage of restaurants offering table booking and online delivery services.
- Compare the average ratings of restaurants with and without table booking.
- Analyze the availability of online delivery options across various price ranges.

**Task 2: Price Range Analysis**

- Determine the most common price range among all restaurants.
- Calculate the average rating for each price range.
- Identify the color associated with the highest average rating, simplifying the identification of top-rated price ranges.

**Task 3: Feature Engineering**

- Leverage advanced feature engineering techniques to extract additional insights from existing columns.
- Create new features, such as "Has Table Booking" and "Has Online Delivery," by encoding categorical variables, enhancing the dataset's intelligence.

The overarching challenge is to enhance our restaurant insights by diving deep into the dataset and uncovering valuable information related to table booking, delivery services, pricing, and feature engineering. This data enrichment will contribute significantly to data-driven decision-making and project success.

# **Task**

# ***Task 1: Table Booking and Online Delivery***

## Import libraries

In [4]:
# Importing Libraries
import pandas as pd
import numpy as np

# Visualization Libraries
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns

# Ignore all warnings
import warnings
warnings.filterwarnings('ignore')

In [8]:
# Load Dataset from github repository
df = pd.read_csv("https://raw.githubusercontent.com/Ansu-Devog/Data-analysis-Internship-Cognifyz-/main/Dataset.csv")

In [1]:
!wget https://raw.githubusercontent.com/Ansu-Devog/Data-analysis-Internship-Cognifyz-/main/Dataset .csv


--2024-05-23 03:51:53--  https://raw.githubusercontent.com/Ansu-Devog/Data-analysis-Internship-Cognifyz-/main/Dataset
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.111.133, 185.199.110.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2024-05-23 03:51:53 ERROR 404: Not Found.

--2024-05-23 03:51:53--  http://.csv/
Resolving .csv (.csv)... failed: Name or service not known.
wget: unable to resolve host address ‘.csv’


## Dataset Load

## Percentage of Table Booking and Online Delivery

In [9]:
# Percentage of restaurants offering table booking and online delivery
# Total number of restaurants
total_restaurants = len(df)

# Percentage calculation
table_booking_percentage = (df['Has Table booking'] == 'Yes').sum() / total_restaurants * 100
online_delivery_percentage = (df['Has Online delivery'] == 'Yes').sum() / total_restaurants * 100

# Display results
print(f"Percentage of restaurants offering Table Booking: {table_booking_percentage:.2f}%")
print(f"Percentage of restaurants offering Online Delivery: {online_delivery_percentage:.2f}%")

Percentage of restaurants offering Table Booking: 12.12%
Percentage of restaurants offering Online Delivery: 25.66%


## Restaurants with and without Table Booking

In [10]:
# Compare average ratings of restaurants with and without table booking
avg_rating_with_table = df[df['Has Table booking'] == 'Yes']['Aggregate rating'].mean()
avg_rating_without_table = df[df['Has Table booking'] == 'No']['Aggregate rating'].mean()

# Display results
print(f"Average rating with Table Booking: {avg_rating_with_table:.2f}")
print(f"Average rating without Table Booking: {avg_rating_without_table:.2f}")

Average rating with Table Booking: 3.44
Average rating without Table Booking: 2.56


## Availability of Online Delivery

In [11]:
# Availability of online delivery among restaurants with different price ranges
# Select price ranges
price_ranges = df['Average Cost for two'].apply(lambda x: 'Low' if x < 500 else 'Medium' if 500 <= x <= 1000 else 'High')
online_delivery_by_price_range = df.groupby(price_ranges)['Has Online delivery'].value_counts(normalize=True).unstack()

# Display results
print("Online Delivery Availability by Price Range:")
print(online_delivery_by_price_range)

Online Delivery Availability by Price Range:
Has Online delivery         No       Yes
Average Cost for two                    
High                  0.719149  0.280851
Low                   0.858523  0.141477
Medium                0.566209  0.433791


## What did i found from the level 2 (task 1)?

* Percentage of restaurants offering Table Booking 12.12% and Online Delivery 25.66%
* Average rating with Table Booking 3.44 and without Table Booking: 2.56
* Online delivery availability for medium price range food product is better compare to high and low price range food product


# ***Task 2: Price Range Analysis***

# Most Common Price Range

In [ ]:
# Determining the most common price range among all the restaurants
most_common_price_range = df['Price range'].mode()[0]

# Display result
print(f"Most Common Price Range: {most_common_price_range}")

NameError: name 'df' is not defined

## Average Rating for Each Price Range

In [ ]:
# Calculating average rating for each price range
# Group by 'Price range' and calculate the average rating
avg_rating_by_price_range = df.groupby('Price range')['Aggregate rating'].mean()

# Display result
print("Average rating for each price range:")
print(round(avg_rating_by_price_range,3))

## Highest Average Rating Among Different Price Ranges

In [ ]:
# Identifying the color that represents the highest average rating among different price ranges
# Find the price range with the highest average rating
highest_avg_rating_color = avg_rating_by_price_range.idxmax()

# Create the bar plot
plt.bar(avg_rating_by_price_range.index, avg_rating_by_price_range, color='yellow')

# Set the color of the bar corresponding to the highest average rating to a distinct color (e.g., red)
plt.bar(highest_avg_rating_color, avg_rating_by_price_range[highest_avg_rating_color], color='red')

# Set labels
plt.xlabel('Price Range')
plt.ylabel('Average Rating')
plt.title('Average Rating by Price Range')

# Highlight the bar for the highest average rating
plt.show()

NameError: name 'avg_rating_by_price_range' is not defined

## What did i found from the level 2 (task 2)?

* Most common price range among all the restaurants is 1
* Price range 4 get the highest average rating, which is 3.818. Followed by price range 3, 2 and 1

# ***Task 3: Feature Engineering***

## Extract Additional Features

In [ ]:
# Extracting additional features from the existing columns, such as the length of the restaurant name or address
# Create a new column for the length of restaurant names
df['Restaurant Name Length'] = df['Restaurant Name'].apply(len)

# Create a new column for the length of restaurant addresses
df['Address Length'] = df['Address'].apply(len)

# Display the updated DataFrame
df.head()

NameError: name 'df' is not defined